copybookreader-plugins/src/main/java/co/cask/hydrator/plugin/batch/CopybookSource.java

```diff
@@ -0,0 +1,251 @@
+    switch (fieldType) {
+      case NULL:
+        return null;
+      case INT:
+        return value.asInt();
+      case DOUBLE:
+        return value.asDouble();
+      case BOOLEAN:
+        return value.asBoolean();
+      case LONG:
+        return value.asLong();
+      case FLOAT:
+        return value.asFloat();
+      case STRING:
+        return value.asString();
+      default:
+        throw new IOException(String.format("Unsupported schema: %s for field: \'%s\'", field.getSchema(),
+                                            field.getName()));
+    }
+  }
+
+  /**
+   * Create schema from list field names retrieved from CopybookFiles
+   *
+   * @param fieldsSet set of field names extracted from Copybook
+   * @return output schema fields
+   */
+  private Schema getSchema(Set<String> fieldsSet) {
+    List<Schema.Field> fields = Lists.newArrayList();
+    for (String field : fieldsSet) {
+      fields.add(Schema.Field.of(field, Schema.nullableOf(Schema.of(Schema.Type.STRING))));
+    }
+    return Schema.recordOf("record", fields);
+  }
+
+  /**
+   * Config class for CopybookSource.
+   */
+  public static class CopybookSourceConfig extends ReferencePluginConfig {
+
+    @Description("Complete path of the .bin to be read; for example: 'hdfs://10.222.41.31:9000/test/DTAR020_FB.bin' " +
+      "or 'file:///home/cdap/DTAR020_FB.bin'.\n " +
+      "This will be a fixed-length binary format file that matches the copybook.\n" +
+      "(This is done to accept files present on a remote HDFS location.)")
+    private String binaryFilePath;
+
+    @Description("Contents of the COBOL copybook file which will contain the data structure. For example: \n" +
+      "000100*                                                                         \n" +
+      "000200*   DTAR020 IS THE OUTPUT FROM DTAB020 FROM THE IML                       \n" +
+      "000300*   CENTRAL REPORTING SYSTEM                                              \n" +
+      "000400*                                                                         \n" +
+      "000500*   CREATED BY BRUCE ARTHUR  19/12/90                                     \n" +
+      "000600*                                                                         \n" +
+      "000700*   RECORD LENGTH IS 27.                                                  \n" +
+      "000800*                                                                         \n" +
+      "000900        03  DTAR020-KCODE-STORE-KEY.                                      \n" +
+      "001000            05 DTAR020-KEYCODE-NO      PIC X(08).                         \n" +
+      "001100            05 DTAR020-STORE-NO        PIC S9(03)   COMP-3.               \n" +
+      "001200        03  DTAR020-DATE               PIC S9(07)   COMP-3. ")
+    private String copybookContents;
+
+    @Nullable
+    @Description("The schema for the data as it will be formatted in CDAP. Sample schema: " +
```

Comment:
I still don't understand why this is being used instead of being derived from the copybook.  Was there a decision made on that?